### PR TITLE
fix: do not reset route after successful user validation after login

### PR DESCRIPTION
**Pull request by Mohammer5**
* Only [this line](https://github.com/dhis2-sre/im-web-client/pull/501/files#diff-4517ee214f2d319010047f74c7efaaba6feed0c59dc9a80894d7a62a421e0b5dL47) was the issue.
* I also address linter issues which made their way to the master branch
* I removed the superfluous `redirectPath` state (because now we're not redirecting to "login" anymore)

### CODE REVIEW

**Review by Chisomchima — APPROVED** [review]
lgtm

File: src/index.tsx [hunk 1]
```diff
@@ -40,14 +40,25 @@ const router = createBrowserRouter(
...
-root.render(
-    <React.StrictMode>
-        <AlertsProvider>
-            <CssReset />
-            <CssVariables colors theme layers spacers elevations />
-            <RouterProvider router={router} />
-            <Alerts />
-        </AlertsProvider>
-    </React.StrictMode>
+function render(tree) {
```
Comment (tonsV2) [inline src/index.tsx:43]: What's the purpose of these changes? And are they related to the original issue?

File: src/components/auth-provider.tsx [hunk 2]
```diff
@@ -17,8 +17,7 @@ export const AuthProvider: FC = () => {
     const navigate = useNavigate()
     const [isAuthenticating, setIsAuthenticating] = useState(false)
     const [authenticationErrorMessage, setAuthenticationErrorMessage] = useState('')
-    const [redirectPath, setRedirectPath] = useState('')
-    const [currentUser, _setCurrentUser] = useState<User | null>(getCurrentUserFromLocalStorage)
+    const [currentUser, _setCurrentUser] = useState<User | null>(() => getCurrentUserFromLocalStorage())
     const setCurrentUser = useCallback((nextUser) => {
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/components/auth-provider.tsx:21]: You mean why we have to use "useCallback"?

In that case:
* The `login` function (wrapped with a `useCallback`) would get created again on every re-render.
* Every hook that has `login` as a dependency would get executed again.
* Also the hook that handles the `UNAUTHORIZED_EVENT` event would get executed on ever render

Wrapping `setCurrentUser` in `useCallback` prevents that from happening.

File: src/index.tsx (hunk 1 above)
Comment (Mohammer5) [reply src/index.tsx:43]: Yes, I was logged out immediately because during dev mode, the strict mode is active and executes every hook twice. This caused some weird issues with hooks that are supposed to run once only.

This effects development mode only and is backwardscompatible, so you'd have to actively set the flag for anything to change. But the strictmode can be quite annoying at times

File: src/components/auth-provider.tsx [hunk 3]
```diff
@@ -34,26 +33,34 @@ export const AuthProvider: FC = () => {
...
-    // Redirect if already logged
-    useEffect(() => {
-        async function checkLoggedIn() {
-            try {
-                const userResponse = await getUser()
-
-                if (currentUser.id !== userResponse.data.id) {
-                  setCurrentUser(userResponse.data)
+    const [checkingUser, setCheckingUser] = useState(true)
+    useEffect(
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/components/auth-provider.tsx:37]: In this case it's on a separate line so all arguments have the same indentation.
The second argument has a comment (disabling eslint to ask the dev to put the dependencies there), so that one needs to be indented like that.

File: src/components/auth-provider.tsx [hunk 4]
```diff
@@ -34,26 +33,34 @@ export const AuthProvider: FC = () => {
...
+        () => {
+            async function checkLoggedIn() {
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply src/components/auth-provider.tsx:39]: If that's an issue? This should be covered by a linting rule. If you insist I can change it though. In the platform team we consider this personal code style, which - when in doubt - should be covered by a linting rule, otherwise it's up to the developer.

[the remark this answers is not in the document]
Comment (tonsV2) [reply src/components/auth-provider.tsx:39]: I very much agree and yes, it should be covered by a linting rule

File: src/components/auth-provider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonsV2) [reply src/components/auth-provider.tsx:21]: Awesome! Thanks for the clarification :slightly_smiling_face:

File: src/index.tsx (hunk 1 above)
Comment (tonsV2) [reply src/index.tsx:43]: Thanks for explaining :pray:

**Review by tonsV2 — APPROVED** [review]
